dcp-0006/dcp-0006.mediawiki

```diff
@@ -0,0 +1,412 @@
+* Risk of coercion being used against key project members.
+* Process of treasury spending is insufficiently transparent.
+
+The proposed solution will semi-automate the release of treasury funds while
+diminishing the influence of key project members over the treasury spends.
+
+==Process==
+
+Currently, every block that is mined must include a coinbase transaction that
+generates a UTXO of a specific amount that requires a 2-of-3 multisig to spend.
+This is enforced by consensus. During payout time, DHG creates and signs
+transactions to all payees. This process is not adequate to support the
+features proposed in this DCP.
+
+Essentially, the treasury is converted from a UTXO based model to an account
+based model. The account balance is determined by starting with an initial
+balance of zero followed by state changes that are recorded on the blockchain.
+This process "burns" and "creates" UTXOs as needed. Sending UTXOs to the
+treasury burns the coins and adds value to the treasury account and spending
+from the treasury creates UTXOs.  This process is analogous to the ticket
+purchases and redeems. The treasury changes shall be added to the staking
+mechanism. The stake tree was selected for these changes because the coin burn
+and creation is analogous to the consensus voting mechanism. Additionally, the
+treasury votes are added as optional data to <code>OP_SSGEN</code> so that
+voting can happen simultaneously.
+
+This DCP introduces three distinct new processes.
+# Subsidy generation for the Treasury is now accomplished by a new <code>treasurybase</code> stake transaction type which MUST be the first entry in the stake transaction tree as opposed to the <code>coinbase</code> of the regular transaction tree.
+# Crediting the Treasury from existing UTXOs is handled by a new <code>treasury add</code> stake transaction type that burns the UTXO(s) and credits the Treasury account accordingly.  This is analogous to purchasing a ticket.
+# Spending treasury funds now requires stakeholders to vote on the expenditure. The Politeia (or Pi) key operators create a <code>treasury spend</code> transaction and sign it.  This transaction is then broadcast onto the network where it enters the <code>mempool</code>.  Voting wallets are expected to detect this transaction and vote <code>Yes</code> or <code>No</code> on the expenditure based on stakeholder preference via the existing <code>vote</code> stake transactions.  The <code>treasury spend</code> transaction is not valid until enough <code>Yes</code> votes are cast by the stakeholders at which point it is included in a block thereby finalizing the payouts.
+
+==Specification==
+
+The following section details the new transaction types.  All new
+transactions shall be part of the stake tree and all treasury opcodes shall be
+rejected if used in scripts in the regular tree.
+
+===Treasurybase===
+
+The newly introduced treasurybase moves the developer subsidy from the coinbase
```

Comment:
`<code>treasurybase</code>` for consistency.

Reply:
Did that but now I feel the urge to wrap coinbase and stuff.